DataDog/datadog-agent
# [CONTP-286] Revert new DD_ENTITY_ID

### What does this PR do?
[CONTP-286](https://datadoghq.atlassian.net/browse/CONTP-286)

1. Revert new DD_ENTITY_ID
2. Remove admission_controller.inject_config.inject_container_name from config 



### Motivation


Solution chosen in the New Origin Detection Spec RFC has been implemented (DD_EXTERNAL_ENV: `it-INIT_CONTAINER,cn-CONTAINER_NAME,pu-POD_UID`). 

Therefore, the DD_ENTITY_ID as  `en-(init.)$(DD_INTERNAL_POD_UID)/$(DD_INTERNAL_CONTAINER_NAME)` from PRs: 
1. https://github.com/DataDog/datadog-agent/pull/23447
3. https://github.com/DataDog/datadog-agent/pull/24482
are no longer needed. 


### Additional Notes



### Possible Drawbacks / Trade-offs



### Describe how to test/QA your changes
unit test updated
e2e test updated
QA same as 
https://github.com/DataDog/datadog-agent/pull/27313 and 
https://github.com/DataDog/datadog-agent/pull/27449
to make sure the new implementation is not broken




[CONTP-286]: https://datadoghq.atlassian.net/browse/CONTP-286?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
